org.OpenT2T.Sample.SuperPopular.Helpers/js/index.js

```diff
@@ -0,0 +1,169 @@
+'use strict';
+
+//var request = require('request');
+var https = require('https');
+var q = require('q');
+
+//this is our base, we each call has its own set of options
+ var options = 
+ {
+        protocol: 'https:',
+        host: 'api.wink.com'     
+ };
+    
+var  accessToken;  //we will need this when creating the headers
+      
+module.exports = 
+{
+    //get the initial parameters to build our target endpoint
+   initWinkApi: function(apiEndpoint, deviceID, token)
+   {
+       options.path = '/' + apiEndpoint + '/' + deviceID;
+       accessToken = token; //the specific token
+        
+   },
+ 
+   sendDesiredStateCommand: function(apiField,value,callback) 
```

Comment:
hmm ok. so should I just remove the callback as a parameter? 

For example in the test.js right now it works kind of like:
```
wh.getLastReading('someparam',function (error, result) //last reading fields are read-only
{
    console.log(result); 
});
```
I'm not familiar with promises, so how would it work in that case using the ".then()"?